### PR TITLE
fix getpileupsummary should not run if no germline resource is provided

### PR DESCRIPTION
**Pull request by FriederikeHanssen**
## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/sarek/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, also make a PR on the nf-core/sarek _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

**Comment by FriederikeHanssen** [comment]
Don't merge yet, running manual test!

**Comment by FriederikeHanssen** [comment]
Results from manual run:

```
gitpod /workspace/sarek (germline_resource) $ PROFILE=docker pytest --tag mutect2_manual --symlink --keep-workflow-wd
======================================================================================================================== test session starts ========================================================================================================================
platform linux -- Python 3.10.12, pytest-7.4.0, pluggy-1.2.0
rootdir: /workspace/sarek
plugins: workflow-2.0.1
collecting ... 
collected 97 items                                                                                                                                                                                                                                                  

Run variant calling on somatic sample with mutect2 without intervals:
        command:   nextflow run main.nf -profile test_cache,tools_somatic --tools mutect2 --no_intervals --outdir results
        directory: /tmp/pytest_workflow_pf6g0wxm/Run_variant_calling_on_somatic_sample_with_mutect2_without_intervals
        stdout:    /tmp/pytest_workflow_pf6g0wxm/Run_variant_calling_on_somatic_sample_with_mutect2_without_intervals/log.out
        stderr:    /tmp/pytest_workflow_pf6g0wxm/Run_variant_calling_on_somatic_sample_with_mutect2_without_intervals/log.err
'Run variant calling on somatic sample with mutect2 without intervals' done.

Run variant calling on somatic sample with mutect2:
        command:   nextflow run main.nf -profile test_cache,tools_somatic --tools mutect2 --outdir results
        directory: /tmp/pytest_workflow_pf6g0wxm/Run_variant_calling_on_somatic_sample_with_mutect2
        stdout:    /tmp/pytest_workflow_pf6g0wxm/Run_variant_calling_on_somatic_sample_with_mutect2/log.out
        stderr:    /tmp/pytest_workflow_pf6g0wxm/Run_variant_calling_on_somatic_sample_with_mutect2/log.err
'Run variant calling on somatic sample with mutect2' done.

Run joint calling on tumor only samples with mutect2:
        command:   nextflow run main.nf -profile test_cache,tools_tumoronly --input tests/csv/3.0/recalibrated_tumoronly_joint.csv --tools mutect2 --joint_mutect2 --outdir results
        directory: /tmp/pytest_workflow_pf6g0wxm/Run_joint_calling_on_tumor_only_samples_with_mutect2
        stdout:    /tmp/pytest_workflow_pf6g0wxm/Run_joint_calling_on_tumor_only_samples_with_mutect2/log.out
        stderr:    /tmp/pytest_workflow_pf6g0wxm/Run_joint_calling_on_tumor_only_samples_with_mutect2/log.err
'Run joint calling on tumor only samples with mutect2' done.

tests/test_tools_manually.yml .................................................................................................                                                                                                                               [100%] Keeping temporary directories and logs. Use '--kwd' or '--keep-workflow-wd' to disable this behaviour.
```

File: conf/test.config [hunk 1]
```diff
@@ -70,21 +70,12 @@ process {
...
-            //sample name from when the test data was generated
-            ext.args = { "--f1r2-tar-gz ${task.ext.prefix}.f1r2.tar.gz --normal-sample normal " }
-        }
+    withName: '.*:MUTECT2_PAIRED'{
+        //sample name from when the test data was generated
+        ext.args = { "--f1r2-tar-gz ${task.ext.prefix}.f1r2.tar.gz --normal-sample normal" }
     }
 
-    withName: 'FILTERVARIANTTRANCHES'{
+    withName: '.*:FILTERVARIANTTRANCHES'{
```
Comment (asp8200) [inline conf/test.config:78]: I see you made a similar change elsewhere in this PR. What is the advantage of adding `.*:`?

Comment (maxulysse) [reply conf/test.config:78]: so that we overwrite the default value in the actual configs

File: workflows/sarek.nf [hunk 2]
```diff
@@ -499,7 +499,7 @@ workflow SAREK {
 
     // Tabix indexed vcf files:
     dbsnp_tbi              = params.dbsnp                   ? params.dbsnp_tbi             ? Channel.fromPath(params.dbsnp_tbi).collect()             : PREPARE_GENOME.out.dbsnp_tbi             : Channel.value([])
-    germline_resource_tbi  = params.germline_resource       ? params.germline_resource_tbi ? Channel.fromPath(params.germline_resource_tbi).collect() : PREPARE_GENOME.out.germline_resource_tbi : Channel.value([])
+    germline_resource_tbi  = params.germline_resource       ? params.germline_resource_tbi ? Channel.fromPath(params.germline_resource_tbi).collect() : PREPARE_GENOME.out.germline_resource_tbi : [] //do not change to Channel.value([]), the check for its existince then fails for Getpileupsumamries
```
Comment (maxulysse) [inline workflows/sarek.nf:502]:
```suggestion
    germline_resource_tbi  = params.germline_resource       ? params.germline_resource_tbi ? Channel.fromPath(params.germline_resource_tbi).collect() : PREPARE_GENOME.out.germline_resource_tbi : [] //do not change to Channel.value([]), the check for its existence then fails for Getpileupsumamries
```

File: conf/test.config (hunk 1 above)
Comment (asp8200) [reply conf/test.config:78]: Thanks for enlighting me on this. I would have expected `withName: 'FILTERVARIANTTRANCHES'` to also overwrite the values from the "actual" configs.

Comment (maxulysse) [reply conf/test.config:78]: `.*` has priority over more specific selectors

**Review by maxulysse — APPROVED** [review]

**Comment by FriederikeHanssen** [comment]
Still waiting for an answer from Thomas. All my tests on real data with his parameters work as intended though

**Pull request merged**